wallets/core/src/wallet.ts

```diff
@@ -182,34 +182,32 @@ class Wallet<InstanceType = any> {
     // Inserting accounts into our state.
     let nextAccounts: string[] = [];
     let nextNetwork: Network | null | undefined = null;
+    let activeEvmNetwork: Network | null = null;
     if (Array.isArray(connectResult)) {
       const accounts = connectResult.flatMap((blockchain) => {
         const chainId = blockchain.chainId || Networks.Unknown;
         // Try to map chainId with a Network, if not found, we use chainId directly.
         const network =
           getBlockChainNameFromId(chainId, this.info.supportedBlockchains) ||
           Networks.Unknown;
+
+        // check if this chain is an evm network
```

Comment:
It can be something like this:

```suggestion
        // When connecting to an evm instance, it will return address and wallet's active chain. On switch network we are comparing state's network and what passed as requestedNetwork. This code is for making sure we are setting active chain in state if it's evm.
```